handlers/node_api.py

```diff
@@ -311,3 +311,65 @@ def graphviz_dot_neighbors(
         "body": response,
         "headers": {"Content-Type": "application/json"},
     }
+
+
+@ApiGateway()
+def nc_state(
+    event: ApiGatewayEvent, _context: LambdaContext, node_api: Optional[NodeApi] = None
+) -> dict:
+    """Get state of a nano contract."""
+    node_api = node_api or NodeApi()
+    id = event.query.get("id")
+    fields = event.multiValueQueryStringParameters.get("fields[]", [])
+    balances = event.multiValueQueryStringParameters.get("balances[]", [])
+    calls = event.multiValueQueryStringParameters.get("calls[]", [])
+
+    if id is None:
+        raise ApiError("invalid_parameters")
+
+    response = node_api.get_nc_state(id, fields, balances, calls)
+    return {
+        "statusCode": 200,
+        "body": json.dumps(response or {}),
+        "headers": {"Content-Type": "application/json"},
+    }
+
+
+@ApiGateway()
+def nc_history(
+    event: ApiGatewayEvent, _context: LambdaContext, node_api: Optional[NodeApi] = None
+) -> dict:
+    """Get history of a nano contract."""
+    node_api = node_api or NodeApi()
+    id = event.query.get("id")
+    after = event.query.get("after")
+    count = event.query.get("count")
+
+    if id is None:
+        raise ApiError("invalid_parameters")
+
+    response = node_api.get_nc_history(id, after, count)
+    return {
+        "statusCode": 200,
+        "body": json.dumps(response or {}),
+        "headers": {"Content-Type": "application/json"},
+    }
+
+
+@ApiGateway()
+def nc_blueprint_information(
+    event: ApiGatewayEvent, _context: LambdaContext, node_api: Optional[NodeApi] = None
+) -> dict:
+    """Get blueprint information."""
+    node_api = node_api or NodeApi()
+    blueprint_id = event.query.get("blueprint_id")
+
+    if blueprint_id is None:
+        raise ApiError("invalid_parameters")
+
+    response = node_api.get_nc_blueprint_information(blueprint_id)
```

Comment:
Should we treat exceptions raised in this call?

We could have a `HathorCoreTimeout` raised here, for instance. 

It's also possible for a JSON-string to be returned by `HathorCoreClient` containing an `error` field, apparently: https://github.com/HathorNetwork/hathor-explorer-service/blob/57cc7294878bf185be743a4dac503a1d4025ccfa/gateways/clients/hathor_core_client.py#L158

Reply:
It makes sense, I added an error handler for when the client class returns `None` or a dict with `error` on it, so we can return `500` to the client. https://github.com/HathorNetwork/hathor-explorer-service/pull/286/commits/2196b506114e2921eac4aacd8e7df13ff69b400f

I also created an issue to improve the error handling in general for the explorer service. https://github.com/HathorNetwork/hathor-explorer-service/issues/319

Regarding the  `HathorCoreTimeout`, it's already raised in the `HathorCoreClient` and handled by the `ApiGateway` decorator class.